generation/diff_tool.sh

```diff
@@ -0,0 +1,112 @@
+#!/bin/bash
+#run this script from google-cloud-java directory
+# google-cloud-java$ ./generation/diff-tool.sh
+# not using set -e because failing diff command is ok
+# was thinking to use staging-directory ID as an argument to this script, but we have different staging ID's for different
+# group ID artifacts so it will complicate the command.
+# correct repositories -> comgooglecloud-5374, comgoogleapi-4857, comgoogleanalytics-1049
```

Comment:
You should probably make these variables at the top of the script so that they can be easily changed for another run.

Reply:
@ddixit14  As per discussion, add instruction on how to retrieve these component IDs and how to supply them to this script. I prefer 3 arguments, but environment variables are also good.

Reply:
Please refer to this [PR](https://github.com/googleapis/google-cloud-java/pull/8251) for latest changes. I forgot that this script will run on a branch which does not have "-SNAPSHOT" in any of the pom.xml's. I have addressed all the concerns in the script comments.